src/middleware/authentication/middleware.ts

```diff
@@ -1,18 +1,37 @@
+// biome-ignore lint/correctness/noNodejsModules: we need the node crypto module
+import { createHash } from "node:crypto";
 import type { NextFunction, Request, Response } from "express";
 import { introspectToken } from "@lib/idp/introspectToken";
+import { RedisConnector } from "@src/lib/redisConnector";
+
+const cacheExpiry = 60; // seconds
+
+export function getAccessTokenCacheKey(accessToken: string): string {
+  const hash = createHash("sha256").update(accessToken).digest("hex");
+  return `api:auth:${hash}`;
```

Comment:
I used a hash here for two reasons:
1. Improve key lookup using the shorter hash value.
2. Avoid storing an access token in Redis that could be used to make authenticated requests on behalf of the user.

Reply:
I really like this idea.  Nice approach.